include/deal.II/matrix_free/fe_evaluation_data.h

```diff
@@ -1680,50 +1652,33 @@ namespace internal
           fu(out[i], array[indices[i] / N][indices[i] % N]);
         }
   }
-} // namespace internal
-
-
-
-template <int dim, typename Number, bool is_face>
-inline Number
-FEEvaluationData<dim, Number, is_face>::read_cell_data(
-  const AlignedVector<Number> &array) const
-{
-  Number out = Number(1.);
-  internal::process_cell_or_face_data(this->get_cell_ids(),
-                                      array,
-                                      out,
-                                      [](auto &local, const auto &global) {
-                                        local = global;
-                                      });
-  return out;
-}
-
-
 
-template <int dim, typename Number, bool is_face>
-inline void
-FEEvaluationData<dim, Number, is_face>::set_cell_data(
-  AlignedVector<Number> &array,
-  const Number &         in) const
-{
-  internal::process_cell_or_face_data(this->get_cell_ids(),
-                                      array,
-                                      in,
-                                      [](const auto &local, auto &global) {
-                                        global = local;
-                                      });
-}
+  template <std::size_t N, typename VectorOfArrayType, typename ArrayType>
+  inline void
+  set_valid_element_to_array(const std::array<unsigned int, N> indices,
+                             const VectorOfArrayType &         array,
+                             ArrayType &                       out)
+  {
+    // set all entries in array to a valid element
```

Comment:
Could you add an assert that `indices` and `out` have the same size?

Reply:
I added an assertion in the last commit.